server/chat-commands/core.ts

```diff
@@ -1526,6 +1526,64 @@ export const commands: ChatCommands = {
 		}
 	},
 
+	requestpartner(target, room, user) {
+		target = this.splitTarget(target);
+		const targetUser = this.targetUser;
+		if (!targetUser) return this.popupReply(`User not found.`);
+		if (targetUser.locked && !user.locked) {
+			return this.popupReply(`That user is locked and cannot be invited to battles.`);
+		}
+		if (user.locked && !targetUser.locked) {
+			return this.errorReply(`You are locked and cannot invite others to battles.`);
+		}
+		const format = Dex.formats.get(target);
+		if (!format.exists) return this.popupReply(`Invalid format: ${target}`);
+		if (format.gameType !== 'multi') {
+			return this.popupReply(`You cannot invite people to non-multibattle formats. Challenge them instead.`);
+		}
+
+		let requestMap = Ladders.requests.get(targetUser.id);
+		if (!requestMap) {
+			requestMap = new Map();
+			Ladders.requests.set(targetUser.id, requestMap);
+		}
+		requestMap.set(user.id, format.id);
+		targetUser.send(`|requestmulti|${user.name}|${format.id}`);
+		this.connection.send(
+			`|pm|${user.getIdentity()}|${targetUser.getIdentity()}|/text` +
+			`You invited ${targetUser.name} to join you in ${format.name}.`
```

Comment:
Shouldn't there be a space either after `text` or before `You`?

Reply:
yes. done.